backend/src/llm_clients/openai_clients_base.py

```diff
@@ -0,0 +1,27 @@
+import abc
+from pydantic import BaseModel, Field
+
+
+class MessagePassingInterface(BaseModel):
+    system_content: str = Field(...)
+    user_content: str = Field(...)
+
+    @property
+    def to_dict(self) -> list[dict[str, str]]:
+        dicts = [
+            {"role": "system", "content": f"{self.system_content}"},
+            {"role": "user", "content": f"{self.user_content}"}
+        ]
+        return dicts
```

Comment:
1. 第 6、7 行的 `Field` 初始化僅傳入 `...`，建議可以直接刪除 `Field`，因為他不是定義資料欄位所需的必要項目。（如果需要調整欄位的別名、預設值⋯⋯，才需要用到 `Field`）
2. 由於此類別的方法 `to_dict()` 有加上 `@property` 裝飾器，可以直接把當作唯獨的屬性存取，建議可以用名詞命名。
3. 此類別的方法 `to_dict()` 實際上是回傳 `list` 而不是 `dict`。
4. 此專案中所有程式初始化 `MessagePassingInterface` 時都直接傳入 `system_content` 和 `user_content`（如 `src/llm_clients/openai_clients.py` 第 38 行）：
```py
MPI=MessagePassingInterface(system_content=system_content, user_content=user_content)
```

看起來繼承 `pydantic.BaseModel` 稍嫌多餘（除非為了資料驗證），這裡可以簡單定義一個普通的 class 就好：

```suggestion
class MessagePassingInterface:

    def __init__(self, system_content: str, user_content: str):
        self.system_content = system_content
        self.user_content = user_content

    @property
    def llm_messages() -> list:
        ...
```